fcrepo_verify/verifier.py

```diff
@@ -0,0 +1,154 @@
+from csv import DictWriter
+import sys
+import time
+import threading
+from rdflib.compare import isomorphic
+
+from fcrepo_verify.constants import EXT_BINARY_EXTERNAL
+from fcrepo_verify.iterators import FcrepoWalker, LocalWalker
+from fcrepo_verify.resources import FedoraResource, LocalResource
+
+
+class FedoraImportExportVerifier:
+    """contains logic for performing a verification"""
+
+    def __init__(self, config, loggers):
+        self.config = config
+        self.loggers = loggers
+
+    def execute(self):
+        """executes the verification process"""
+        config = self.config
+        csv = self.config.csv
+
+        loggers = self.loggers
+        logger = loggers.file_only
+        console = loggers.console
+        console_only = loggers.console_only
+
+        # Set up csv file, if specified
+        if csv:
+            csvfile = open(csv, "w")
+            fieldnames = ["number", "type", "original", "destination",
+                          "verified",
+                          "verification"]
+            writer = DictWriter(csvfile, fieldnames=fieldnames)
+            writer.writeheader()
+
+        console.info("Starting verification...")
+        if config.mode == "export":
+            tree = FcrepoWalker(config, logger)
+        elif config.mode == "import":
+            tree = LocalWalker(config, logger)
+
+        console.info("Running verification on Fedora 4 {0}"
+                     .format(config.mode))
+
+        success_count = 0
+        failure_count = 0
+
+        def total_count():
+            return success_count + failure_count
+
+        def log_summary(logger):
+            logger.info(
+                "Verified {} resources:  successes = {},  failures={}"
```

Comment:
for consistency, should be
`... successes = {}, failures = {}`
(uniform spacing)